### PR TITLE
[CALCITE-3251] BinaryExpression evaluate method support full numeric types in Primitive

### PR DESCRIPTION
**Pull request by xy2953396112**
In BinaryExpression evaluate method，maybe we should support Long type.
When I run a function with a return value of long, this function adds a constant. The following error will be thrown.
image

### CODE REVIEW

File: linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java [hunk 1]
```diff
@@ -61,6 +61,8 @@ public Object evaluate(Evaluator evaluator) {
       case DOUBLE:
         return (Double) expression0.evaluate(evaluator)
                + (Double) expression1.evaluate(evaluator);
+      case LONG:
+        return (Long) expression0.evaluate(evaluator) + (Long) expression1.evaluate(evaluator);
```
Comment (hsyuan) [inline linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java]: Can you wrap the line just like L62~63 does? This is Calcite convention.

File: linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java [hunk 2]
```diff
@@ -97,7 +103,9 @@ public Object evaluate(Evaluator evaluator) {
       case DOUBLE:
         return (Double) expression0.evaluate(evaluator)
                >= (Double) expression1.evaluate(evaluator);
-      default:
+      case LONG:
+        return (Long) expression0.evaluate(evaluator) >= (Long) expression1.evaluate(evaluator);
+        default:
```
Comment (hsyuan) [inline linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java]: The indent of `default`

File: linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java [hunk 3]
```diff
@@ -108,7 +116,9 @@ public Object evaluate(Evaluator evaluator) {
       case DOUBLE:
         return (Double) expression0.evaluate(evaluator)
                < (Double) expression1.evaluate(evaluator);
-      default:
+      case LONG:
+        return (Long) expression0.evaluate(evaluator) < (Long) expression1.evaluate(evaluator);
+        default:
```
Comment (hsyuan) [inline linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java]: indent

**Comment by hsyuan** [comment]
The tests failed, please check the style.

**Comment by xy2953396112** [comment]
> I clicked `Approve` by mistake. Sorry about that.

Ok,I have fixed the indent.Could you review again?

**Comment by xy2953396112** [comment]
> The tests failed, please check the style.

please help me review the test failed.

**Comment by xy2953396112** [comment]
> The tests failed, please check the style.

Those tests have been passed，please review again.

**Comment by danny0405** [comment]
@xy2953396112  I re-trigger the tests and they all passed.

**Comment by xy2953396112** [comment]
> @xy2953396112 I re-trigger the tests and they all passed.

THX,is there any problem?

**Comment by danny0405** [comment]
@xy2953396112 The changes looks good, thanks,

**Comment by danny0405** [comment]
@xy2953396112 Can we fix for all the exactly numeric types and fractional types ?
- For integers: `BYTE` to `BIGINT`)
- For fractional: `REAL`, `FLOAT`, `DOUBLE`
- For decimal: `DECIMAL`s

**Comment by xy2953396112** [comment]
> @xy2953396112 Can we fix for all the exactly numeric types and fractional types ?
> 
> * For integers: `BYTE` to `BIGINT`)
> * For fractional: `REAL`, `FLOAT`, `DOUBLE`
> * For decimal: `DECIMAL`s

ok,could we merge it firstly,and i will commit a new pr

> @xy2953396112 Can we fix for all the exactly numeric types and fractional types ?
> 
> * For integers: `BYTE` to `BIGINT`)
> * For fractional: `REAL`, `FLOAT`, `DOUBLE`
> * For decimal: `DECIMAL`s

ok，could we merge it firstly ? and i will commit a new PR later for numeric types and fractional types.

**Comment by danny0405** [comment]
@xy2953396112 Thanks, let's fix them in one PR but with different commits.

**Comment by xy2953396112** [comment]
> @xy2953396112 Thanks, let's fix them in one PR but with different commits.

@danny0405 @hsyuan please review.

File: linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java [hunk 4]
```diff
@@ -58,9 +58,18 @@ public Object evaluate(Evaluator evaluator) {
       case INT:
         return (Integer) expression0.evaluate(evaluator) + (Integer) expression1
             .evaluate(evaluator);
+      case BYTE:
+        return (Byte) expression0.evaluate(evaluator) + (Byte) expression1
+            .evaluate(evaluator);
```
Comment (danny0405) [inline linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java]: How about the `SHORT` type ? I think you should cover all the numeric types defined in Primitive.java[1].

[1] https://github.com/apache/calcite/blob/a68bfc96bacc95b4b8ab5d185bb5cc1991113769/linq4j/src/main/java/org/apache/calcite/linq4j/tree/Primitive.java#L44

Comment (xy2953396112) [reply linq4j/src/main/java/org/apache/calcite/linq4j/tree/BinaryExpression.java]: Thanks for you review.This code have been cover the SHORT type.

**Pull request closed without merging**